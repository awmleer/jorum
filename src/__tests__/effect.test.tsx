import * as React from 'react'
import {BehaviorSubject, interval} from 'rxjs'
import {bloc, checkStream, effect, Provider, suspense, useBloc, useStream} from '..'
import {sleep} from './utils'
import {map} from 'rxjs/operators'
import {render} from 'react-testing-library'

it('effect is executed', async function () {
  @bloc
  class TestBloc {
    readonly data$ = interval(50)
    readonly test$ = new BehaviorSubject('no')
    
    @effect
    private readonly handleDataChange$ = this.data$.pipe(
      map(() => {
        this.test$.next('yes')
      })
    )
  }
  
  const Show = suspense(() => {
    const testBloc = useBloc(TestBloc)
    const test = useStream(testBloc.test$)
    checkStream()
    return (
      <div>
        {test}
      </div>
    )
  })
  
  const renderer = render(
    <Provider of={TestBloc}>
      <Show />
    </Provider>
  )
  
  await sleep(100)
  expect(renderer.asFragment()).toMatchSnapshot()
  renderer.unmount()
})
