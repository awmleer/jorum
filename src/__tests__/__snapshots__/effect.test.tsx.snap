// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`effect is executed 1`] = `
<DocumentFragment>
  <div>
    yes
  </div>
</DocumentFragment>
`;
